core/src/main/scala/org/apache/spark/security/ShellBasedGroupsMappingProvider.scala

```diff
@@ -36,10 +36,12 @@ private[spark] class ShellBasedGroupsMappingProvider extends GroupMappingService
     userGroups
   }
 
-  // shells out a "bash -c id -Gn username" to get user groups
+  // shells out a "id -Gn username" to get user groups
   private def getUnixGroups(username: String): Set[String] = {
-    val cmdSeq = Seq("bash", "-c", "id -Gn " + username)
+    val idPath = Utils.executeAndGetOutput("which" :: "id" :: Nil).stripLineEnd
```

Comment:
Why do we need a separate `which` command? Why can't we just invoke `id` directly?

Reply:
We can do .. I wanted to keep the behavior as much as possible (e.g., bash might point a different `id`) .. but probably I was overthinking. Directly invoking `id` would be better.

Reply:
Are you concerned about whether the `id` command might resolve differently in bash vs. the default shell? That's fair: it might be unlikely to be different, but maybe we want to play it safe.

What do you think about caching the value in a `private lazy val` right before this method so we only invoke it once?